wicket-core/src/main/java/org/apache/wicket/markup/html/form/upload/resource/FileUploadToResourceField.js

```diff
@@ -0,0 +1,99 @@
+        this.clientSideCancelCallBack = clientSideCancelCallBack;
+        this.connectionErrorCallBack = connectionErrorCallBack;
+    }
+
+    Wicket.FileUploadToResourceField.prototype.upload = function()
+    {
+        // we add the files to a FormData object.
+        var formData = new FormData();
+        var totalfiles = this.input.files.length;
+        for (var index = 0; index < totalfiles; index++) {
+            formData.append("FILE-UPLOAD",this.input.files[index]);
+        }
+        // pass the input name to knw where to store files at server side.
+        formData.append("uploadId", this.inputName);
+        var self = this;
+        // we use jQuery to post the file to the resource (this.resourceUrl)
+        // and we keep a reference to the request in order to be able
+        // to cancel the upload
+        this.xhr = $.ajax({
+            url: this.resourceUrl,
+            type: "POST",
+            data: formData,
+            processData: false,
+            contentType: false,
+            success: function (res) {
+                // do clean up on success
+                self.clientSideSuccessCallBack();
+                if (res.error) {
+                    var ep = {'success': false, 'errorMessage': res.errorMessage};
+                    Wicket.Ajax.get({"u": self.ajaxCallBackUrl, "ep": ep});
+                    self.connectionErrorCallBack(res);
+                } else {
+                    var ep = {'success': true, 'filesInfo': JSON.stringify(res)};
+                    Wicket.Ajax.get({"u": self.ajaxCallBackUrl, "ep": ep});
+                }
+            },
+            error: function (jqXHR, textStatus, errorThrown) {
+                if (textStatus === "abort") {
+                    // user aborted the upload.
+                    self.clientSideCancelCallBack();
+                    var ep = {'success': false, 'errorMessage': 'upload.canceled'};
+                    Wicket.Ajax.get({"u": self.ajaxCallBackUrl, "ep": ep});
+                } else if (textStatus === "error"){
+                    var ep = {'success': false, 'errorMessage': errorThrown};
+                    Wicket.Ajax.get({"u": self.ajaxCallBackUrl, "ep": ep});
+                    self.connectionErrorCallBack();
+                } else if (textStatus === "parsererror"){
+                    // this error will only happen is generated JSON at server side is faulty
+                    var data = jqXHR.responseText;
+                    Wicket.Log.log(data);
+                }
+            }
+        });
+    }
+
+    // cancel the upload
+    Wicket.FileUploadToResourceField.prototype.cancel = function () {
+        // we have a reference to the request we can cancel it.
+        if (this.xhr) {
+            this.xhr.abort();
+            Wicket.Log.log("upload canceled!");
+            delete (this.xhr);
+        } else {
+            Wicket.Log.log("Too late to cancel: upload already finished.");
+        }
```

Comment:
For easier debugging I'd suggest to use ` this.inputName ` in the log messages above.

Reply:
Done.